live-build/misc/migration-scripts/dx_apply

```diff
@@ -139,6 +139,10 @@ zfs create \
 	"$RPOOL/ROOT/$FSNAME/home" ||
 	die "failed to create linux dataset $RPOOL/ROOT/$FSNAME/home"
 
+#
+# Note that this dataset should not contain any useful data as it will be
```

Comment:
Are you sure the order of operations here is correct? i.e. to extract the contents of the archive first, then effectively delete anything the archive might have put in /var/delphix, and replace it with a clone..

For linux upgrade, we create the clone first, then update/install the new packages. So, if the package placed things in `/var/delphix`, they'd remain there. Whereas, I think in this implementation, if the package placed things in `/var/delphix`, they'd be removed.

Would it be better to first create the clone, then extract the archive? I know you state that you need to create the clone in `dx_execute`, but I'm wondering if that's actually "safe" to do.. Can we at least add some verification to ensure the linux `/var/delphix` dataset is empty before it's destroyed and replaced?

Reply:
The dataset is not actually fully empty, it contains the empty "dropbox" directory. I can add some verifications to make sure that there isn't anything else in there though. The dataset definitely has to be cloned in `dx_execute` as configuration in `/var/delphix/server/etc/` and just logs in general would be out of date if we do this in `dx_apply`.

Reply:
@prakashsurya I've done some refactoring so that the dataset is checked and destroyed in `dx_apply`. Could you take another look?